### PR TITLE
Cleanup exceptions to at least use RuntimeException and include previous

### PR DESCRIPTION
**Pull request by dereuromark**
CakePHP should never throw just the most generic exception here, but instead a bit more precise.
Fixing it to at least Runtime for now, as it is already in other similar places.

Also fixed previous to be included where forgotten and fixed error code to expected int type.

### CODE REVIEW

**Comment by dereuromark** [comment]
Failing phpstan must be a defect on master?
```
 ------ ----------------------------------------------------------------------- 
  Line   Http/ResponseEmitter.php                                               
 ------ ----------------------------------------------------------------------- 
  240    Parameter #3 $expires of function setcookie expects int, array given.  
 ------ ----------------------------------------------------------------------- 
```

**Comment by ADmad** [comment]
> Failing phpstan must be a defect on master?

It's a defect in phpstan, it doesn't recognize the new signature of `setcookie()` added in PHP 7.3 even though it's run on PHP 7.3.

File: src/Core/InstanceConfigTrait.php [hunk 1]
```diff
@@ -296,7 +297,7 @@ protected function _configDelete(string $key): void
 
         foreach ($stack as $i => $k) {
             if (!is_array($update)) {
-                throw new Exception(sprintf('Cannot unset %s value', $key));
+                throw new RuntimeException(sprintf('Cannot unset %s value', $key));
```
Comment (ADmad) [inline src/Core/InstanceConfigTrait.php]: This is an unnecessary change. The exception here is `\Cake\Core\Exception\Exception` which extends `\RuntimeException`, it's not PHP's `\Exception`.

File: src/View/SerializedView.php [hunk 2]
```diff
@@ -127,7 +127,7 @@ function ($v) {
             } catch (Exception | TypeError $e) {
                 throw new SerializationFailureException(
                     'Serialization of View data failed.',
-                    null,
+                    0,
```
Comment (ADmad) [inline src/View/SerializedView.php]: This should stay `null` as the `Cake\Core\Exception\Exception` sets the error code to value of it's `$_defaultCode` property and if the argument is `null`.

File: src/Core/InstanceConfigTrait.php (hunk 1 above)
Comment (dereuromark) [reply src/Core/InstanceConfigTrait.php]: Ah, you are right about this.

File: src/View/SerializedView.php (hunk 2 above)
Comment (dereuromark) [reply src/View/SerializedView.php]: Ah, makes sense then.
